Feature/ratings create

rripken opened the pull request.

rma-rripken committed:
Moved getRatings method out of CwmsDataManager and into RatingDao.  Added RatingDao interface. MetricRegistry already does a getOrCreate internally so we don't need to hold meter/timer references.

rma-rripken committed:
Replaced time_context with timeContext

rma-rripken committed:
Need hec-cwms-ratings and hec-data

rma-rripken committed:
Merge branch 'develop' into feature/ratings_create

# Conflicts:
#	cwms_radar_api/src/main/java/cwms/radar/api/LevelsController.java

MikeNeilson commented on cwms_radar_api/src/main/java/cwms/radar/data/dao/RatingSetDao.java:
```diff
@@ -0,0 +1,125 @@
+		}
+		catch(DataAccessException ex)
+		{
+			throw new IOException("Failed to create Rating");
```
This should be "retrieve rating."

rripken replied:
agreed

MikeNeilson reviewed (changes requested):
Everything else looks good. Just that one I assume copy paste mishap with the exception message text.

rma-rripken committed:
Added an overridable method deserializeFromXml to simplify testing.

rma-rripken committed:
Extracted a configureRoutes method so that tests can optionally setup their own routes with mocks or spys.

rma-rripken committed:
RatingControllerTest will need cwms-rating

rma-rripken committed:
Added a test that verifies that create with xml forwards to deserializeFromXml.

rma-rripken committed:
Merge branch 'develop' into feature/ratings_create

rma-rripken committed:
Stop the RadarAPI in a finally block.  There is probably a better way to do this.

rma-rripken committed:
Minor typos

rma-rripken committed:
A memory leak was aparrently fixed in javalin 4.1.1.  Also logging checkstyle version to try and debug the gradle update issue.

rma-rripken committed:
Merge branch 'develop' into feature/ratings_create

rripken committed:
Merge branch 'develop' into feature/ratings_create

rma-rripken committed:
Added util class for interacting with RatingSet to/from json.  Added a getOne for fetching RatingSet in new v2 formats.  Added support for create/update in JOSNv2.

rma-rripken committed:
Merge branch 'develop' into feature/ratings_create

# Conflicts:
#	cwms_radar_standalone/src/test/java/mil/army/usace/hec/RatingsControllerTest.java

rma-rripken committed:
Updated test

rma-rripken committed:
Including xsl resources.

rma-rripken committed:
Throwing an exception if a specified resource is not found.

rripken commented:
@MikeNeilson Do you need to do something to approve your change request from before?  I fixed that a while ago but I just added some of the json support so you probably should review that too.

MikeNeilson commented on build.gradle:
```diff
@@ -10,7 +10,7 @@ ext.dropwizard_version="4.1.2"
 ext.JAXB_API_VERSION="2.2.12"
 ext.JAXB_IMPL_VERSION="2.2.11"
-ext.JAVALIN_VERSION='4.0.1'
+ext.JAVALIN_VERSION='4.1.1'
```
I very much appreciate that we are actively keeping this up-to-date instead of lingering.

MikeNeilson commented on cwms_radar_api/src/main/java/cwms/radar/api/Controllers.java:
```diff
@@ -0,0 +1,18 @@
+package cwms.radar.api;
```
This was a great idea, btw.

MikeNeilson commented on cwms_radar_standalone/src/main/java/mil/army/usace/hec/RadarAPI.java:
```diff
@@ -168,29 +168,34 @@ public RadarAPI(javax.sql.DataSource ds, int port) throws Exception {
-            crud("/pools/{pool-id}", new PoolController(metrics));
-        });
 
+        .routes( () -> configureRoutes());
```
good change, that Javalin setup section is getting rather large.
